get_element_cpo/P2769.bs

```diff
@@ -686,13 +669,20 @@ inline constexpr auto get_value = get_element&lt;1>;
 
 <ins>
 <blockquote>
-<span>1. The name <code>get_element</code> denotes a customization point object (<a href="https://eel.is/c++draft/customization.point.object">[<b>customization.point.object</b>])</a>.
-    The expression <code>get_element&lt;I>(E)</code> where <code>I</code> is <code>size_t</code> for a subexpression <code>E</code> is
-    expression-equivalent to:
-    1. <code>get&lt;I>(E)</code>, if <code>E</code> has class or enumeration type and <code>get&lt;I>(E)</code> is a well-formed expression
-        when treated as an unevaluated operand, where the meaning of <code>get</code> is established as-if by
-        performing argument-dependent lookup only (<a href="https://eel.is/c++draft/basic.lookup.argdep">[<b>basic.lookup.argdep]</b>)</a>.
-    2. Otherwise, <code>get_element&lt;I>(E)</code> is ill-formed.</span>
+The name <code>get_element</code> denotes a customization point object (<a href="https://eel.is/c++draft/customization.point.object">[<b>customization.point.object</b>])</a>.
+     The expression <code>get_element&lt;I>(E)</code> where <code>I</code> is <code>size_t</code> for a subexpression
+     <code>E</code> is expression-equivalent to:
+1. <code>E.get&lt;I>()</code> if <code>E</code> has class or enumeration type <code>T</code>, which has non-static
+     member function template <code>get</code> and <code>E.get&lt;I>()</code> is a well-formed expression.
+1. Otherwise, <code>get&lt;I>(E)</code>, if <code>E</code> has class or enumeration type and <code>get&lt;I>(E)</code>
+     is a well-formed expression when treated as an unevaluated operand, where the meaning of <code>get</code>
+     is established as-if by performing argument-dependent lookup only
+     (<a href="https://eel.is/c++draft/basic.lookup.argdep">[<b>basic.lookup.argdep]</b>)</a>.
+1. Otherwise, <code>get_element&lt;I>(E)</code> is ill-formed.
+     If for type <code>T</code> of expression <code>E</code> the
+     <code>typename tuple_size&lt;remove_cvref_t&lt;T>>::type</code> expression is ill-formed,
+     <code>get_element&lt;I>(E)</code> is ill-formed
+
```

Comment:
```suggestion
The name <code>get_element</code> denotes a customization point object (<a href="https://eel.is/c++draft/customization.point.object">[<b>customization.point.object</b>])</a>.
     Given <code>I</code> an integral constant of type <code>size_t</code> and a subexpression E <code>E</code> with type <code>T</code>:

  - If  <code> tuple_size_t&lt;remove_cvref_t&lt;T>> </code> does not denote a valid type, 
    <code>get_element&lt;I>(E)</code> is ill-formed.
  - Otherwise, if  <code>E.get&lt;I>()</code> is a is a valid expression,  <code>get_element&lt;I>(E)</code> 
    is expression equivalent to <code>E.get&lt;I>()</code>
  -  Otherwise,  if <code>E</code> has class type and <code>get&lt;I>(E)</code> is a valid expression where the meaning of <code>get</code>
     is established as-if by performing argument-dependent lookup only
     (<a href="https://eel.is/c++draft/basic.lookup.argdep">[<b>basic.lookup.argdep]</b>)</a>,  <code>get_element&lt;I>(E)</code>  is expression equivalent to <code>E.get&lt;I>()</code>
   - Otherwise, <code>get_element&lt;I>(E)</code> is ill-formed.

```

Should we constrain the return value of get_element? (based of tuple_element_t)

Reply:
Thanks for you suggestions. I applied them all. Two things here, though:
- I don't know if we should constrain the return value of `get_element` to be `tuple_element_t`. I would leave it as is right now.
- I fix the wording to use `typename tuple_size<T>::type` because there is no `tuple_size_t`.